crates/sovereign-sdk/full-node/sov-stf-runner/src/runner.rs

```diff
@@ -649,51 +646,41 @@ where
         let mut height = self.start_height;
         info!("Starting to sync from height {}", height);
 
-        let mut last_connection_error = Instant::now();
-        let mut last_parse_error = Instant::now();
-
-        let mut connection_index = 0;
-        let mut retry_index = 0;
-
         loop {
-            let soft_batch = match self
-                .sequencer_client
-                .get_soft_batch::<Da::Spec>(height)
-                .await
-            {
-                Ok(Some(soft_batch)) => soft_batch,
-                Ok(None) => {
-                    debug!(
-                        "Soft Batch: no batch at height {}, retrying in {} seconds",
-                        height, RETRY_SLEEP
-                    );
-                    Self::log_error(
-                        &mut last_parse_error,
-                        RETRY_INTERVAL,
-                        &mut retry_index,
-                        "No soft batch published".to_string().as_str(),
-                    );
-                    sleep(Duration::from_secs(RETRY_SLEEP)).await;
-                    continue;
-                }
-                Err(e) => match e.downcast_ref::<jsonrpsee::core::Error>() {
-                    Some(Error::Transport(e)) => {
-                        debug!("Soft Batch: connection error during RPC call: {:?}", e);
-                        Self::log_error(
-                            &mut last_connection_error,
-                            CONNECTION_INTERVALS,
-                            &mut connection_index,
-                            format!("Soft Batch: connection error during RPC call: {:?}", e)
-                                .as_str(),
-                        );
-                        sleep(Duration::from_secs(RETRY_SLEEP)).await;
-                        continue;
-                    }
-                    _ => {
-                        anyhow::bail!("Soft Batch: unknown error from RPC call: {:?}", e);
+            let exponential_backoff = ExponentialBackoffBuilder::new()
+                .with_initial_interval(Duration::from_secs(60))
+                .with_max_elapsed_time(Some(Duration::from_secs(15 * 60)))
+                .build();
+            let inner_client = &self.sequencer_client;
+            let soft_batch = backoff::future::retry(exponential_backoff, || async move {
+                match inner_client.get_soft_batch::<Da::Spec>(height).await {
+                    Ok(Some(soft_batch)) => Ok(soft_batch),
+                    Ok(None) => {
+                        debug!("Soft Batch: no batch at height {}, retrying...", height);
+                        Err(backoff::Error::Transient {
+                            err: "No soft batch published".to_owned(),
```

Comment:
if we are in this match arm, we shouldnt backoff exponentially

should just be 2 seconds

Reply:
Set a constant 2 second delay.